src/main/g8/src/main/scala/$package__packaged$/JokeAlg.scala

```diff
@@ -0,0 +1,48 @@
+package $package$
+
+import cats.Applicative
+import cats.effect.Sync
+import cats.implicits._
+import io.circe.{Encoder, Decoder, Json, HCursor}
+import org.http4s.{EntityDecoder, EntityEncoder, Method, Uri, Request}
+import org.http4s.client.Client
+import org.http4s.circe._
+
+trait JokeAlg[F[_]]{
+  def getJoke: F[JokeAlg.Joke]
+}
+
+object JokeAlg {
+  final case class Joke(joke: String) extends AnyVal
+  object Joke {
+    // We can make this easier if we use circe-generic
```

Comment:
Then why didn't we?  I think we should do it or explain why not.

I fear that this JSON boilerplate makes a bad first impression, though there is an API compatibility argument for it. :man_shrugging:

Reply:
Or we could also use `circe-derivation`, which keeps the encoder creation, but makes them way more reasonable.

Reply:
I think that has the same problem: if the case class changes, the serialization format changes.  You either have to manually write your codecs, or sign in blood that you won't change the case classes in your HTTP tier and hope the default doesn't ever change, or compromise by going manual after the first change.

Another problem: circe-derivation is for circe-0.9, and we're on 0.10 here.